internal/analytics/client/async/client.go

```diff
@@ -91,9 +94,14 @@ func (a *Client) Wait() {
 }
 
 func (a *Client) sendEvent(category, action, label string, dims ...*dimensions.Values) error {
-	if a.closed {
+	switch {
+	case a.closed:
 		logging.Debug("Client is closed, not sending event")
 		return nil
+	case (a.cfg.IsSet(constants.ReportAnalayticsConfig) && !a.cfg.GetBool(constants.ReportAnalayticsConfig)) ||
+		strings.ToLower(os.Getenv(constants.DisableAnalyticsEnvVarName)) == "true":
+		logging.Debug("Analytics are disabled; not sending event")
+		return nil
```

Comment:
This is introducing new behavior, and is causing the config to be evaluated on every analytical event which slows down execution. I don't think we need this either because the async client reports into the sync client, which already has this logic.

Reply:
Thanks for letting me know that async reports to sync. That's good to know.